test/dymaptic.GeoBlazor.Core.Test.Blazor.Shared/Components/FeatureLayerTests.razor

```diff
@@ -470,4 +470,361 @@
         await AssertJavaScript("assertLayerExists", args: "tile");
         await AssertJavaScript("assertLayerExists", args: "feature");
     }
+
+    [TestMethod]
+    public async Task FeatureLayer_BasicPropertyRoundtrip_And_JS_Asserts(Action renderHandler)
+    {
+        FeatureLayer? layer = null;
+
+        AddMapRenderFragment(
+            @<MapView class="map-view" OnViewRendered="renderHandler">
+                <Map>
+                    <Basemap>
+                        <BasemapStyle Name="BasemapStyleName.ArcgisTopographicBase" />
+                    </Basemap>
+                    <FeatureLayer @ref="layer"
+                                  Source="@([])"
+                                  OutFields="@(["*"])"
+                                  SpatialReference="SpatialReference.Wgs84"
+                                  GeometryType="FeatureGeometryType.Point"
+                                  ObjectIdField="OBJECTID" />
+                </Map>
+            </MapView>);
+
+        await WaitForMapToRender();
+        await AssertJavaScript("assertLayerExists", args: "feature");
+
+        await layer!.SetLegendEnabled(false);
+        Assert.AreEqual(false, await layer.GetLegendEnabled());
+        await AssertJavaScript("featureLayerAsserts.assertFeatureLayerBooleanPropEquals", args: [layer!.Id, "legendEnabled", false]);
```

Comment:
Looking back at this file now, we could simplify and move all of these to `assertObjectHasPropertyWithValue`.